Microsoft.Packaging.Tools/tasks/Trimming/NuGetPackageNode.cs

```diff
@@ -32,37 +30,21 @@ public NuGetPackageNode(string id, string version, IEnumerable<string> dependenc
         public IEnumerable<NuGetPackageNode> Dependencies { get { return _dependencies; } }
         public IList<FileNode> Files { get; } = new List<FileNode>();
 
-        public void PopulateDependencies(Dictionary<string, NuGetPackageNode> allPackages, ILog log)
+        public void AddDependency(NuGetPackageNode dependencyNode)
         {
-            foreach(var dependencyId in _dependencyIds)
-            {
-                NuGetPackageNode dependencyNode;
-                if (!allPackages.TryGetValue(dependencyId, out dependencyNode))
-                {
-                    // package declared a dependency but NuGet was missing the dependent package
-                    // in the lock file.  This indicates a broken restore, but don't fail trimming
-                    log.LogMessage(LogImportance.Low, $"Could not locate dependency {dependencyId} of package {Id}.");
-                }
-                else
-                {
-                    _dependencies.Add(dependencyNode);
+            _dependencies.Add(dependencyNode);
 
-                    // Runtime packages may be brought in by a file-based dependency,
-                    // but runtime packages may be missing the dependencies needed since those are 
-                    // often declared by the idenity package since it is in the compile graph
-                    // and capable of bringing in other runtime-split packages.
+            // Runtime packages may be brought in by a file-based dependency,
+            // but runtime packages may be missing the dependencies needed since those are 
+            // often declared by the idenity package since it is in the compile graph
+            // and capable of bringing in other runtime-split packages.
 
-                    // Map back up to the identity package so that we can root it and its dependencies.
-                    // This creates an artificial cycle, but our graph walk doesn't care about cycles.
-                    if (dependencyNode.IsRuntimePackage)
-                    {
-                        dependencyNode._dependencies.Add(this);
-                    }
-                }
+            // Map back up to the identity package so that we can root it and its dependencies.
+            // This creates an artificial cycle, but our graph walk doesn't care about cycles.
+            if (dependencyNode.IsRuntimePackage)
+            {
+                dependencyNode._dependencies.Add(this);
```

Comment:
It's just a heuristic.  Worst case it will end up rooting some stuff up-stack.  If we get that feedback we can change it.